### PR TITLE
Preserve the value of isClpCompliantApp on resolved url in odsp

### PR DESCRIPTION
**Pull request by jatgarg**
(https://dev.azure.com/fluidframework/internal/_workitems/edit/422/)

For more information about how to contribute to this repo, visit this [page](https://github.com/microsoft/FluidFramework/blob/main/CONTRIBUTING.md)

## Description

Preserve the value of isClpCompliantApp on resolved url in odsp in create new case as it was not getting persisted. Its value was not set on the header of the request to be resolved so it was getting lost.

## PR Checklist

> Use the check-list below to ensure your branch is ready for PR. If the item is not applicable, leave it blank.

-   [ ] I have updated the documentation accordingly.
-   [x] I have added tests to cover my changes.
-   [x] All new and existing tests passed.
-   [x] My code follows the code style of this project.
-   [x] I ran the lint checks which produced no new errors nor warnings for my changes.
-   [ ] I have checked to ensure there aren't other open Pull Requests for the same update/change.

## Does this introduce a breaking change?

-   [ ] Yes
-   [x] No

> If this introduces a breaking change, please describe the impact and migration path for existing applications below.

### CODE REVIEW

File: packages/drivers/odsp-driver/src/test/createNewUtilsTests.spec.ts [hunk 1]
```diff
@@ -204,4 +195,39 @@ describe("Create New Utils Tests", () => {
...
+                async () => createNewFluidFile(
+                    async (_options) => "token",
+                    newFileParams,
+                    new TelemetryNullLogger(),
+                    createSummary(),
+                    epochTracker,
+                    fileEntry,
+                    true,
+                    false,
+                    true,
```
Comment (markfields) [inline packages/drivers/odsp-driver/src/test/createNewUtilsTests.spec.ts:226]:
```suggestion
                    true /* isClpCompliantApp */,
```

Comment (markfields) [reply packages/drivers/odsp-driver/src/test/createNewUtilsTests.spec.ts:226]: And also explicitly validate that the value is false if that param is missing or `false`? (maybe in an existing test that doesn't set it)

File: packages/drivers/odsp-driver/src/createFile.ts [hunk 2]
```diff
@@ -91,7 +93,10 @@ export async function createNewFluidFile(
 
     const odspUrl = createOdspUrl({ ... newFileInfo, itemId, dataStorePath: "/" });
     const resolver = new OdspDriverUrlResolver();
-    const odspResolvedUrl = await resolver.resolve({ url: odspUrl });
+    const odspResolvedUrl = await resolver.resolve({
+        url: odspUrl,
+        headers: { [ClpCompliantAppHeader.isClpCompliantApp]: isClpCompliantApp },
```
Comment (markfields) [inline packages/drivers/odsp-driver/src/createFile.ts:98]: If `isClpCompliantApp` is `undefined` should this header get the value `undefined` or `false`?

Comment (markfields) [reply packages/drivers/odsp-driver/src/createFile.ts:98]: Also, existing usage in `epochTracker` does `toString()`, feels like the two places should match.

Comment (jatgarg) [reply packages/drivers/odsp-driver/src/createFile.ts:98]: Both will have the same effect as long as it is not true.

File: packages/drivers/odsp-driver/src/test/createNewUtilsTests.spec.ts (hunk 1 above)
Comment (jatgarg) [reply packages/drivers/odsp-driver/src/test/createNewUtilsTests.spec.ts]: done. Added in same test.

File: packages/drivers/odsp-driver/src/createFile.ts (hunk 2 above)
Comment (jatgarg) [reply packages/drivers/odsp-driver/src/createFile.ts:98]: No, The one in epoch tracker is added in headers of network apis while here it is just set in the resolved url.

**Review by markfields — APPROVED** [review]

**Pull request merged**